core/src/main/java/org/jdbi/v3/core/internal/AnnotationFactory.java

```diff
@@ -59,6 +60,10 @@ private AnnotationFactory() {}
                 return "@" + annotationType.getName() + "()";
             }
 
+            if (method.getDeclaringClass() == annotationType) {
+                return method.getDefaultValue();
```

Comment:
`Annotation` javadocs define an explicit contract for how `equals()` and `hashCode()` should be implemented. Qualifier lookup and comparison will likely see subtle bugs for annotations with default attributes if we fail to follow it--especially considering that `QualifiedType` stores qualifiers in `HashSet`s.

* `Annotation.equals()` returns true if the argument is an annotation, `other.annotationType()` equals `this.annotationType()`, and all attributes are `equals()` to each other.
* `Annotation.hashCode()` returns the sum of attribute hashes, where an attribute hash equals `(127 * attributeName.hashCode()) ^ attributeValue.hashCode()`.
* `Annotation.toString()` doesn't have a strict contract, but we should follow suit with what the JDK does to make troubleshooting more approachable.

The current implementation already complies with the Annotation contract--which was easy since we don't yet allow attributes.

Reply:
Thanks for catching this.  The simple implementation fooled me into thinking it was not important, not sure how I missed this obvious bit.  Fixed.